unicode.cpp

```diff
@@ -197,18 +197,14 @@ static std::unordered_map<std::string, uint8_t> unicode_utf8_to_byte_map() {
     return map;
 }
 
-static inline std::wstring unicode_wstring_from_utf8(const std::string & s)
-{
+static inline std::wstring unicode_wstring_from_utf8(const std::string & s) {
     std::wstring_convert<std::codecvt_utf8<wchar_t>> conv;
     return conv.from_bytes(s);
 }
 
-static inline std::string unicode_wstring_to_utf8(const std::wstring & ws)
-{
-    // code to convert from utf32/utf16 to utf8
-    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> converter;
-    std::string utf8 = converter.to_bytes(ws);
-    return utf8;
+static inline std::string unicode_wstring_to_utf8(const std::wstring & ws) {
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> conv;
+    return conv.to_bytes(ws);
```

Comment:
@dragnil1 Not sure if this is the intent, but the following change of this function makes the tokenizer tests pass on my Mac. Do you think this is OK to change?

Reply:
This change converts **UCS-2** or **UCS-4/UTF-32** encoded `std::wstring` to **UTF-8** encoded `std::string` and the previous one, converts **UTF-16** encoded `std::wstring` to **UTF-8** encoded `std::string` according to [reference](https://en.cppreference.com/w/cpp/locale/wstring_convert). Both works on Ubuntu(tested) but I am not sure about windows as it uses **UTF-16** encoded `std::wstring`.